### ISSUE TITLE
how to update existing video?

### ISSUE THREAD

**Issue by baoang** [WWBN/AVideo-Encoder#362]
Hi,

I'm currently running encoder v3.8 and avideo v10.5.

Recently found accidentally  encoder supports video replacing function. My guess is that if one of the videos was considered not good enought, site admin could re-post/share from the encoder and make the original video replaced.

With my assumption, I tried with one of my prev shared/encoded video by conducting a new encoding process, just like I did with other previously uploaded/shared ones, the only unique step was to fill in the **Update Existing Video** module within **Advanced** section/tab the ID of my prev video(video to be replaced).

However, when the process was over, I found the original video untouched, but a new video was produced and could be found on the video list.

If I wanted to make the desired happen, ie. to replace it rather than re-produce a new one, what should I do then?

**Comment by DanielnetoDotCom** [comment]
This feature was made by somebody else. I will review it to see if is working properly.

however, you can use the swap videos feature

![image](https://user-images.githubusercontent.com/18009293/112722682-ba5b4580-8ee9-11eb-90ca-85a6ef2d33f8.png)

**Comment by akhilleusuggo** [comment]
This feature is powerfull ( specially the one on the encoder ) . The problem is that does not obey the streamer site's configuration .

I have enable swap videos only for admin , but the encoder does show it to everybody .

**Comment by swimitup** [comment]
Hi, indeed the **update_video_id** does nothing. Here's an extract starting line 18 from the main.js located at view/jquery-file-upload/js/ :

```
...
    $('#fileupload').fileupload({
        // Uncomment the following to send cross-domain cookies:
        //xhrFields: {withCredentials: true},
        url: 'view/jquery-file-upload/server/php/?PHPSESSID=' + PHPSESSID,
        maxChunkSize: 5000000, // 5 MB
        add: function (e, data) {
            var videos_id = $('#update_video_id').val();
            var that = this;
            if (videos_id) {
                swal({
                    title: "You will overwrite the video ID: " + videos_id,
                    text: "The video will be replaced with this new file, are you sure you want to proceed?",
                    icon: "warning",
                    buttons: true,
                    dangerMode: true,
                })
                        .then(function (confirm) {
...

```
You were supposed to receive an alert from the swal function confirming if you want to replace the video, but this alert is not given, encoder proceeds with encoding and the uploaded video does not appear in the videos list...

Updated the encoder files right now.

By the way, there's an advancedCustom variable related to this process which you cannot find anywhere in aVideo advanced customize plugin: **doNotAllowUpdateVideoId**.

Line 469 from file view/index.php from encoder:

```
...
                                    if (empty($advancedCustom->doNotAllowUpdateVideoId)) {
                                        ?>
                                        <div class="panel panel-default">
                                            <div class="panel-heading"><i class="fas fa-desktop"></i> Update existing video</div>
                                            <div class="panel-body">
                                                <img id="inputNextVideo-poster" src="view/img/notfound.jpg" class="ui-state-default img img-responsive" alt="">
                                                <input type="text" class="form-control" id="videoSearch" name="videoSearch" placeholder="Search for a video">
                                                <input type="number" class="form-control" id="update_video_id" name="update_video_id" placeholder="Video Id">
                                            </div>
                                        </div>
...

```

**Comment by swimitup** [comment]
Update existing video works fine now with encoder scripts update (git pull)

**Comment by baoang** [comment]
didn't find that *doNotAllowUpdateVideoId* plugin really.

just did a git pull to make encoder up-to-date, but the function still add a new video rather than replace the old one.

or perhaps there are some other places/options to change? I've browsed the plugin of AdvancedCustomize with no clues.

**Comment by DanielnetoDotCom** [comment]
@baoang you were right, it was missing.

I just sent an update

**Comment by baoang** [comment]
> @baoang you were right, it was missing.
> 
> I just sent an update

Thank you, @DanielnetoDotCom , I git pulled and tried again.

Now, the setting is like this, unchecking *doNotAllowUpdateVideoold*
[Imgur](https://imgur.com/DDF30Ux)
![CustomizedAdvanced](https://i.imgur.com/DDF30Ux.png)

Then with encoder in Advanced tab, filling in the video ID, say 124
[Imgur](https://imgur.com/uJA3ipw)
![encoder filling video id](https://i.imgur.com/uJA3ipw.png)

The expected result is, video 124 is replaced by the new video, but it is not, and in AVideo platform, a new video is added.

Both AVideo and Encoder have been updated to the latest version, by `git pull` and `git fetch --all && git reset --hard origin/master`

**Comment by DanielnetoDotCom** [comment]
Hi, thanks for noticing it. I guess I found the bug

update the encoder and streamer and try again.

**Comment by baoang** [comment]
> Hi, thanks for noticing it. I guess I found the bug
> 
> update the encoder and streamer and try again.

Thank you, @DanielnetoDotCom , I've git pulled both encoder and streamer sites.

The search old video blank now works ok. By typing in some key words there the whole title can prompt out.

But the replacing job still not works. The process of replacing surely starts (as the following snapshot indicates) but the old video was not replaced and it produced a new video clip.

[![cekgG6.png](https://z3.ax1x.com/2021/04/02/cekgG6.png)](https://imgtu.com/i/cekgG6)

**Comment by DanielnetoDotCom** [comment]
Hi, did you update the streamer and the encoder?

I have tested here and it works fine

**Comment by baoang** [comment]
> Hi, did you update the streamer and the encoder?
> 
> I have tested here and it works fine

Yes, both streamer and encoder sites are git pulled to the latest.

The git-pull response is as follows(I did it about ten hrs ago, and did it again about five min ago)

![streamer update info](https://z3.ax1x.com/2021/04/02/cmhODH.png)

the following two log files are streamer's and encoder's. I've emptied them before doing another round of replacing old videos, so here are the entire records of replacing the video of 148, but it actually produced 150, the new one, and video 148 remained intact.

[encoder.log](https://github.com/WWBN/AVideo-Encoder/files/6250052/encoder.log)
[avideo.log](https://github.com/WWBN/AVideo-Encoder/files/6250053/avideo.log)

**Comment by baoang** [comment]
an update for this problem

after another try, I found import video (from say Youtube) could not replace the existing video, while upload video (from Basic - From File - Upload) could successfully make the replacement happen.

**Comment by DanielnetoDotCom** [comment]
Hi, sorry but I checked your logs and I could not see any error on it,

**Comment by baoang** [comment]
> Hi, sorry but I checked your logs and I could not see any error on it,

Thank you @DanielnetoDotCom , for keeping up with the issue and previous update of the codes.

Although I couldn't update the existing videos via Youtube source, I could now update that successfully by uploading the videos to the encoder. This method is more controllable from user's side.
